network/wsNetwork.go

```diff
@@ -922,19 +967,29 @@ func (wn *WebsocketNetwork) sendFilterMessage(msg IncomingMessage) {
 func (wn *WebsocketNetwork) broadcastThread() {
 	defer wn.wg.Done()
 	var peers []*wsPeer
+	slowWritingPeerCheckTicker := time.NewTicker(wn.slowWritingPeerMonitorInterval)
+	defer slowWritingPeerCheckTicker.Stop()
 	for {
 		// broadcast from high prio channel as long as we can
 		select {
 		case request := <-wn.broadcastQueueHighPrio:
 			wn.innerBroadcast(request, true, &peers)
 			continue
+		case <-slowWritingPeerCheckTicker.C:
+			wn.checkSlowWritingPeers()
+			continue
+		case <-wn.ctx.Done():
+			return
```

Comment:
Having this `select` be just a single case statement with a default was a slightly useful performance optimization (go compiles a single-case select with a default into a more efficient non-blocking receive, instead of compiling it to the general-purpose `selectgo`).  Can we have the `slowWritingPeerCheckTicker.C` case just in the blocking select below?

Reply:
I'm surprised that this is the case here( no pun intended ). A select statement with a default is equivalent to testing each of the cases sequentially.
but, yes, I think it would be ok if this was previously identified as a botteneck.